app/stac_api/models.py

```diff
@@ -128,6 +129,20 @@ def validate_item_properties_datetimes_dependencies(
             raise ValidationError(_(message))
 
 
+def validate_datetime_string(datetime_string):
+    '''
+    Check if a given string is a datetime string in the expected format or
+    raise an error, if not.
+
+    Raises:
+        ValueError
+    '''
+    try:
+        datetime.strptime(datetime_string, '%Y-%m-%dT%H:%M:%SZ')
```

Comment:
2020-12-14T08:00:00+00:00 is also valid. Please use here the helper function `utils.fromisoformat()` which handle correctly the timezone.